fix(RDBMS customisation): Add new grant required by MySQL 8.0

ugaston opened the pull request:
Version: 7.9

Add the new grant required by MySQL 8.0 to support XA transactions.

ugaston committed:
fix(RDBMS customisation): Add new grant instruction required by MySQL 8.0

tbouffard commented on md/database-configuration.md:
````diff
@@ -237,6 +237,16 @@ See [in-doubt xact resolution Server Configuration Option](https://msdn.microsof
+To support XA transactions, starting from MySQL 8.0, special XA rights must be granted to the database user. For example, if the users for the engine and BDM databases are `bonita` and `business_data` respectively, you should run the following command:
+
+```sql
+GRANT XA_RECOVER_ADMIN ON *.* to bonita, business_data;
````
⚠️ we may add warnings about this because this grant is for all objects.
Could we restrict that?

ugaston replied:
Good point.
I added a warning message. (See latest commit.)

tbouffard replied:
👍

ugaston committed:
Add FLUSH command

As suggested by Marielle, 'FLUSH PRIVILEGES' should be performed after the GRANT for it to be effective straightaway.

ugaston committed:
Consider Thomas' comment.

Take Thomas' comment and add a warning about the scope of the suggested command.

tbouffard commented on md/database-configuration.md:
````diff
@@ -237,6 +237,22 @@ See [in-doubt xact resolution Server Configuration Option](https://msdn.microsof
+```
+
+::: warning
+Please note that the above command, by specifying **`*.*`**, is granting the right for all the databases. You may want to limit the command to your engine and BDM databases if there are additional databases in your MySQL server.
````
```suggestion
Please note that the above command, by specifying **`*.*`**, is granting privileges for all databases. You may want to limit them to the Bonita BPM and BDM databases if there are additional databases in your MySQL server.
```

educhastenier commented on md/database-configuration.md:
````diff
@@ -237,6 +237,22 @@ See [in-doubt xact resolution Server Configuration Option](https://msdn.microsof
+To support XA transactions, starting from MySQL 8.0, special XA rights must be granted to the database user. For example, if the users for the engine and BDM databases are `bonita` and `business_data` respectively, you should run the following command:
+
+```sql
+GRANT XA_RECOVER_ADMIN ON *.* to bonita, business_data;
````
I don't understand... I am convinced I *did* this contribution already, when I discovered this new grant requirement for 7.9 .... :thinking: ...

ugaston commented on md/database-configuration.md:
```diff
@@ -237,6 +237,22 @@ See [in-doubt xact resolution Server Configuration Option](https://msdn.microsof
 
+##### XA Transactions
+
+To support XA transactions, starting from MySQL 8.0, special XA rights must be granted to the database user. For example, if the users for the engine and BDM databases are `bonita` and `business_data` respectively, you should run the following command:
```
```suggestion
To support XA transactions, starting from MySQL 8.0, special XA rights must be granted to the database user. For example, if the users for the Bonita BPM and BDM databases are `bonita` and `business_data` respectively, you should run the following command:
```

ugaston committed:
Apply suggestions from code review

Apply Thomas' suggestion and improve the wording in a previous sentence accordingly.

Co-Authored-By: Thomas Bouffard <thomas.bouffard@bonitasoft.com>

tbouffard reviewed (approved):
LGTM
@ugaston feel free to merge if ok for you
